BlazorMarkDownAppJwt/BlazorMarkDownAppJwt/Server/Controllers/MarkDownController.cs

```diff
@@ -0,0 +1,176 @@
+﻿using BlazorMarkDownAppJwt.Server.Services.MarkDowns;
+using BlazorMarkDownAppJwt.Shared;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using System.Net;
+
+namespace BlazorMarkDownAppJwt.Server.Controllers
+{
+    [ApiController]
+    public class MarkDownController : Controller
+    {
+        private IDocumentService markDownService { get; }
+
+        public MarkDownController(IDocumentService markDownService)
+        {
+            this.markDownService = markDownService;
+        }
+
+        [HttpGet]
+        [Route("api/markdown")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> Get([FromQuery] long currentId)
+        {
+            try
+            {
+
+                var markDownModel = new MarkDownModel
+                {
+                    Body = string.Empty,
+                };
+
+                var document = await markDownService.GetDocument(currentId);
+
+                if (!string.IsNullOrWhiteSpace(document?.MarkDown))
+                {
+
+                    markDownModel.Body = document.MarkDown;
+                    markDownModel.Id = document.Id;
+                }
+
+                return Ok(markDownModel);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
+            }
+
+        }
+
+        [HttpGet]
+        [Route("api/markdown/readme/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(MarkDownModel))]
+        public async Task<IActionResult> GetReadMe()
+        {
+            try
+            {
+
+                var markDownModel = new MarkDownModel
+                {
+                    Body = string.Empty,
+                };
+
+                var document = await markDownService.GetReadMeDocument();
+
+                if (document?.MarkDown != null)
+                {
+
+                    markDownModel.Body = document.MarkDown;
+                }
+
+                return Ok(markDownModel);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);
+            }
+
+        }
+
+        [HttpGet]
+        [Route("api/markdowns/")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(List<MarkDownModel>))]
+        public async Task<IActionResult> GetAll()
+        {
+            try
+            {
+
+                var documents = await markDownService.GetAllDocuments();
+
+                var markdowns = documents?.Select(d => new MarkDownModel
+                {
+                    Id = d.Id,
+                    Body = d.MarkDown,
+                }).ToList();
+
+
+
+                return Ok(markdowns);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError, ex);

```

Comment:
**Remark 3**
 Not needed, ASP.NET Core will generate an internal error 500 by default if you let the exception bubble.
It will also allow tools like Application Insights to "log" crashes in you app.